pkg/registry/core/serviceaccount/strategy.go

```diff
@@ -76,9 +80,17 @@ func (strategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) fie
 
 // WarningsOnUpdate returns warnings for the given update.
 func (strategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
-	return nil
+	return warnIfHasEnforceMountableSecretsAnnotation(ctx, obj.(*api.ServiceAccount))
 }
 
 func (strategy) AllowUnconditionalUpdate() bool {
 	return true
 }
+
+func warnIfHasEnforceMountableSecretsAnnotation(ctx context.Context, serviceAccount *api.ServiceAccount) []string {
```

Comment:
don't warn if oldServiceAccount is non-nil and already has the annotation (if the request isn't newly setting the annotation)

```suggestion
func warnIfHasEnforceMountableSecretsAnnotation(ctx context.Context, serviceAccount, oldServiceAccount *api.ServiceAccount) []string {
```